### ISSUE TITLE
Questions about excuting test cases and database

### ISSUE THREAD

**Issue by OYang007** [Data-to-Insight-Center/komadu#15]
Hi,@isurues 
After executing test cases through web service client,i would like to ask you some questions ,

1. After executing test cases three times, every time the XML graphs were not same. There were not “Non-Context Activity Graph” at the second time. Why is it ?

2. Is provenance information model of Komadu different from Karma’s ?

3. The one is about Karma’s data sets “nosiyGBXML”. Could you tell me something about processing them  and how i should use them? 

Thanks for your reading, your reply will be really appreciated.

Sincerely.
OYang

**Comment by isurues** [comment]
Hi,

Here are are answers to your questions.

1. XML graphs may have elements in different orders if you execute multiple times. But there's no reason for missing “Non-Context Activity Graph”. There should be some exception on the server side. Please check the tomcat log.

2. Yes, Karma was using OPM provenance model. Komadu is using the new W3C PROV specification. 
https://www.w3.org/TR/prov-dm/

3. I'm not familiar with the karma data set that you've mentioned. Could you please share a link to it so that I can have a look?

Isuru

**Comment by OYang007** [comment]
Hi, @isurues 
Thanks for your reply !
Here is a link to data set. [https://pti.iu.edu/impact/data-sets/gsd.html](url)

I think maybe it is OPM graph. I had tried to process it by Komadu visualization tool, but  i failed. So now i have no idea about using it .

Sincerely,
OYang

**Comment by isurues** [comment]
Yes, that is a collection of OPM provenance graphs generated for a set of simulated workflow runs. Komadu visualization tool doesn't support OPM. It only supports W3C PROV graphs. So you won't be able to use the Komadu tool for those OPM graphs. Are you trying to use those OPM graphs for any particular reason?

Isuru

**Comment by OYang007** [comment]
Hi, @isurues 
I try  to use these data to generate  visual provenance graphs, and  test the existing  algorithms, then maybe have follow-up study.

Thanks for your reading!

sincerely,
OYang

**Comment by AhmetGurbuzz** [comment]
Hi isurues,
I have read the komaduuserguide.docx, and in page 13, "5.	You can have a look at the test code in the following file to get an idea on how to write a client for Komadu to ingest notifications and execute query operations.
{komadu_checkout_path}/axis2-client-core/src/test/java/edu/indiana/d2i/komadu/axis2/client/KomaduTests.java  " i saw this. Now, In order to use komadu, Do we need write a client from using Komadutest.java?

**Comment by AhmetGurbuzz** [comment]
@isurues or I misunderstood something? I am 3th grade student and newbie this subjects. 
Sincerely..

**Comment by AhmetGurbuzz** [comment]
By the way, I have solved a problem which is depency problem. Here is the picture of the solution.
![2](https://user-images.githubusercontent.com/19242970/36641985-d422b6c8-1a49-11e8-8886-93ce4d3b4fea.png)
